pkg/reconciler/v1alpha1/route/reconcile_resources.go

```diff
@@ -121,6 +124,10 @@ func (c *Reconciler) reconcilePlaceholderService(ctx context.Context, route *v1a
 		c.Recorder.Eventf(route, corev1.EventTypeNormal, "Created", "Created service %q", name)
 	} else if err != nil {
 		return err
+	} else if !metav1.IsControlledBy(service, route) {
+		// Surface an error in the route's status, and return an error.
+		route.Status.MarkServiceNotOwned(name)
+		return fmt.Errorf("Route: %q does not own Service: %q", route.Name, name)
```

Comment:
should we cascade the same check for the underlying entities of Route, such as ClusterIngress and VirtualService?

Reply:
Yes, and PodAutoscaler and HPA.